lib/tasks/scrape.rake

```diff
@@ -1,146 +1,159 @@
-    # post_description = html_doc.css(data_post_description)
-    # puts "DESC " + post_description.count.to_s
-
-    # data_orig_url = "div.Pin > div.pinWrapper > div.pinImageActionButtonWrapper > a"
-    # orig_url = html_doc.css(data_orig_url)
-    # puts "ORIG_URL " + orig_url.count.to_s
-    # puts orig_url
-
-    # data_category = "div > div > div > a > div.creditTitle"
-    # category = html_doc.css(data_category)
-    # puts "CATEGORY " + category.count.to_s
+    newPin.img_url = picture_url[0]['src']
+    puts "IMG URL " + newPin.img_url
+
+    # not using repins in fritter at the moment
+    # data_repin_count = "meta[name=\"pinterestapp:repins\"]"
+    # repin_count = html_doc.css(data_repin_count)
+    # if not repin_count.any?
+    #   puts "no repins yet"
+    # else 
+    #   puts "REPINS " + repin_count[0]['content'] 
+    # end
+
+    data_like_count = "meta[name=\"pinterestapp:likes\"]"
+    like_count = html_doc.css(data_like_count)
+    #puts like_count
+    if not like_count.any?
+      newPin.like_count = 0
+      puts "no likes yet"
+    else 
+      newPin.like_count = like_count[0]['content']
+      puts "LIKES " + newPin.like_count.to_s
+    end
 
-    # data_number_of_pins = "div.PinBase.ajax.Pin.Module.summary > div.pinWrapper > div.pinMeta > div.pinSocialMeta > a.socialItem > em.socialMetaCount.repinCountSmall"
-    # number_of_pins = html_doc.css(data_number_of_pins)
-    # puts "#PINS " + number_of_pins.count.to_s
+    # some descriptions mention Pinterest quite a lot
+    # I wonder if it's ok to replace that with 'fritter'..?
+    data_description = "meta[name=\"description\"]"
+    description = html_doc.css(data_description)
+    #puts description
+    newPin.optional_info = description[0]['content'].gsub(/Pinterest/, "fritter")
+    puts "DESC " + newPin.optional_info
+    puts "--------------------"
+    puts newPin.inspect
+    puts "--------------------"
+    newPin.save
+    puts "PIN SAVED"
+    puts "--------------------"
+  end
 
-    # data_number_of_likes = "div.PinBase.ajax.Pin.Module.summary > div.pinWrapper > div.pinMeta > div.pinSocialMeta > a.socialItem.likes > em.socialMetaCount.likeCountSmall"
-    # number_of_likes = html_doc.css(data_number_of_likes)
-    # puts "#LIKES " +number_of_likes.count.to_s
+  task :get_categories => :environment do
+    require 'open-uri'
+    require 'nokogiri'
 
+    url = "https://www.pinterest.com/categories/"
+    get_cats(url)
+  end
 
+  def get_cats(url) 
+    document = open(url).read
+    html_doc = Nokogiri::HTML(document)
+    
+    puts "--------------------"
+    puts "IN PAGE " + url
+
+    data_category_list = "a.categoryLinkWrapper"
+    category_list = html_doc.css(data_category_list)
+
+    category_list.each do |item|
+      cat_name = item.css("div.name").inner_text
+      new_cat = Category.create(link: item['href'], name: cat_name)
```

Comment:
Should create a variable for `item['href']`. Just like how you did it with `cat_name`. It will be a lot easier for others to understand. I think the variable name `cat_name` should be `category_name`. Variables should be very easy to understand.